cookbooks/aws-parallelcluster-install/recipes/byos.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster-install
+# Recipe:: byos
+#
+# Copyright 2013-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# setup the user accounts
+include_recipe "aws-parallelcluster-install::user"
+
+# create e.g. /opt/parallelcluster/byos
+byos_base_path = File.join(node['cluster']['base_dir'], node['cluster']['byos']['name'])
+directory byos_base_path do
+  owner node['cluster']['byos']['user']
+  group node['cluster']['byos']['user']
+  mode '0755'
+  action :create
+end
+
+# create e.g. /opt/parallelcluster/shared/byos
+byos_shared_path = File.join(node['cluster']['shared_dir'], node['cluster']['byos']['name'])
+directory byos_shared_path do
+  owner node['cluster']['byos']['user']
+  group node['cluster']['byos']['user']
+  mode '0755'
+  action :create
+  recursive true
```

Comment:
rather than allowing recursive creation I'd make sure `node['cluster']['shared_dir']` is created in base.rb where the other directories are created so that byos does not own it.